src/storage/config.rs

```diff
@@ -54,9 +54,11 @@ impl Default for Config {
 
 impl Config {
     pub fn validate(&mut self) -> Result<(), Box<Error>> {
-        if self.data_dir != DEFAULT_DATA_DIR {
-            self.data_dir = try!(config::canonicalize_path(&self.data_dir))
+        if self.data_dir.is_empty() {
```

Comment:
where is the check? I just remove the default data dir and no compile error occurs.